mantle/kola/harness.go

```diff
@@ -213,6 +214,37 @@ type KoletResult struct {
 const KoletExtTestUnit = "kola-runext"
 const KoletRebootAckFifo = "/run/kolet-reboot-ack"
 
+// Records failed tests for reruns
+type protectedTestResults struct {
+	results []*harness.H
+	mu      sync.RWMutex
+}
+
+func (p *protectedTestResults) add(h *harness.H) {
+	p.mu.Lock()
+	p.results = append(p.results, h)
+	p.mu.Unlock()
+}
+
+func (p *protectedTestResults) getRerunnable() []string {
```

Comment:
I kind of prefer keeping this generic `getTestResults()` and having the logic for filtering them (along with the useful comment you added about why) outside of this interface. WDYT?

Reply:
I felt that maybe the `runProvidedTests` function was becoming too bloated and that this change moved all the logic associated with what it means for a test to be considered "rerunnable" into one place. What I can do instead is keep the generic `getTestResults`, but also add a another function `getRerunnable`. So the results from `getTestResults` would go into `getRerunnable`. Would that be an acceptable compromise?

Reply:
Yeah I guess that wouldn't hurt. Would `getRerunnable()` be part of the `protectedTestResults` interface? I was thinking it was a bit cleaner to keep that interface generic (i.e. it could be used for other things in the future).

Reply:
Nope it would not be a part of the `protetectedTestResults` interface.

Reply:
sounds good